resources/assets/lib/stores/channel-store.ts

```diff
@@ -196,13 +214,26 @@ export default class ChannelStore implements DispatchListener {
     }
   }
 
+  private cleanupIgnoredChannels() {
+    const now = new Date();
```

Comment:
wouldn't it be better to remove the entry on `apiPartChannel` done callback instead of having this clean up?

otherwise at least this should check for map length first before creating new date object.

Reply:
Messages sent to public channels sent after the you leave but before the request finishes processing and arrive after the request finishes processing due to latency, should not cause the channel to reappear in the channel list. 

The whole temporarily tracking channels to ignore messages from, is essentially a workaround for things that aren't currently sending the necessary `chat.channel.join` event. The value of `ignoreInterval` is such that it's beyond any reasonable latency.

Reply:
> Messages sent to public channels sent after the you leave but before the request finishes processing and arrive after the request finishes processing due to latency, should not cause the channel to reappear in the channel list.

This should fail on the `getChannel` request though?

Reply:
Also, only the tab where the leaving is initiated calls `apiPartChannel` 👀

Reply:
Well, no other tab is getting the ignored channel list anyway either

Reply:
Other tabs will get the part channel event relayed to them and ignore the channels.
Sending the join event is the most reliable method of determining whether a message should be dropped or not

Reply:
I'm not sure I follow here. I'm only suggesting instead of `cleanupIgnoredChannels()`, do `apiPartChannel(channelId, core.currentUserOrFail.id).done(() => this.ignoredChannels.delete(channelId))`.

Any messages coming through when it's in process should still be ignored and any excess messages after the request is completed should also still be ignored by 403 error when getting the channel info.